src/main/java/ch/njol/skript/lang/Expression.java

```diff
@@ -85,19 +89,33 @@ default Optional<T> getOptionalSingle(Event e) {
 	 * <p>
 	 * Do not use this in conditions, use {@link #check(Event, Checker, boolean)} instead.
 	 * 
-	 * @param e The event
+	 * @param event The event the expression is being used in.
 	 * @return An array of values of this expression which must neither be null nor contain nulls, and which must not be an internal array.
 	 */
-	public T[] getArray(final Event e);
+	public T[] getArray(Event event);
+	
+	/**
+	 * Get all the values of this expression and then convert them using the provided getter for easy of access.
+	 * 
+	 * @param <G> The returning type of the converter.
+	 * @param event The event the expression is being used in.
+	 * @param converter The converter to do the convertering to the type.
+	 * @return The values completed by the converter.
```

Comment:
The Javadocs should mention that getArray is used, for example with `{@link #getArray(Event) all the values of this expression}` instead of just the text.

They should also mention that the returned array will not be null, and will only contain null values if the converter returned null values.